src/v/kafka/server/materialized_partition.h

```diff
@@ -34,7 +36,7 @@ class materialized_partition final : public kafka::partition_proxy::impl {
         return raft::details::next_offset(_log.offsets().dirty_offset);
     }
 
-    bool is_leader() const final { return true; }
+    bool is_leader() const final { return _partition->is_leader(); }
```

Comment:
this makes sense. why did a previous commit have this hard coded as true?

Reply:
Michal wanted the creation of the API to be in a separate commit. In that commit I couldn't have had is_leader() implemented for materialized topics without changing its constructor to take the extra arg which would have brought in other changes not related to just `adding is_leader to partition_proxy API`